Content.Client/Lobby/UI/LobbyCharacterPreviewPanel.cs

```diff
@@ -141,29 +141,26 @@ public void UpdateUI()
         public static void GiveDummyJobClothes(EntityUid dummy, HumanoidCharacterProfile profile)
         {
             var protoMan = IoCManager.Resolve<IPrototypeManager>();
-
             var entMan = IoCManager.Resolve<IEntityManager>();
-            var inventory = entMan.GetComponent<ClientInventoryComponent>(dummy);
+            var invSystem = EntitySystem.Get<ClientInventorySystem>();
 
             var highPriorityJob = profile.JobPriorities.FirstOrDefault(p => p.Value == JobPriority.High).Key;
 
             // ReSharper disable once ConstantNullCoalescingCondition
             var job = protoMan.Index<JobPrototype>(highPriorityJob ?? SharedGameTicker.FallbackOverflowJob);
 
-            inventory.ClearAllSlotVisuals();
-
-            if (job.StartingGear != null)
+            if (job.StartingGear != null && invSystem.TryGetSlots(dummy, out var slots))
             {
                 var gear = protoMan.Index<StartingGearPrototype>(job.StartingGear);
 
-                foreach (var slot in AllSlots)
+                foreach (var slot in slots)
                 {
-                    var itemType = gear.GetGear(slot, profile);
+                    var itemType = gear.GetGear(slot.Name, profile);
                     if (itemType != string.Empty)
                     {
+                        invSystem.TryUnequip(dummy, slot.Name, true, true);
                         var item = entMan.SpawnEntity(itemType, MapCoordinates.Nullspace);
-                        inventory.SetSlotVisuals(slot, item);
-                        entMan.DeleteEntity(item);
+                        invSystem.TryEquip(dummy, item, slot.Name, true, true);
```

Comment:
`TryEquip` (or more specifically `TryGetSlotContainer`) seems to fail for the inventory dummy?
My lobby preview characters all have no clothing.

Reply:
Ah they don't have a `ContainerManagerComponent`.

`InventorySystem.TryGetSlotContainer()` will create a container if it doesn't exist, but won't actually add a `ContainerManagerComponent` if the mob doesn't have one.

Reply:
fixed it